samza-core/src/main/java/org/apache/samza/zk/ZkLeaderElector.java

```diff
@@ -105,7 +91,7 @@ public boolean tryBecomeLeader() {
     if (index == 0) {
       isLeader.getAndSet(true);
       LOGGER.info(zLog("Eligible to become the leader!"));
-      zkLeaderElectorListener.onBecomingLeader(); // inform the caller
+      leaderElectorListener.onBecomingLeader(); // inform the caller
```

Comment:
Why not make leaderElectorListener a member variable of the LeaderElector instance , instead of an argument to this method? The onBecomingLeader callback is not going to change within the same ZkLeaderElector instance.